lib/prima_auth0_ex.ex

```diff
@@ -8,17 +8,17 @@ defmodule PrimaAuth0Ex do
   @doc """
   Obtain a token for the given audience.
   """
-  @spec token_for(String.t()) :: {:ok, String.t()} | {:error, any()}
-  def token_for(audience) do
-    TokenProvider.token_for(TokenProvider, audience)
+  @spec token_for(atom(), String.t()) :: {:ok, String.t()} | {:error, any()}
+  def token_for(client \\ :default_client, audience) do
```

Comment:
It looks conceptually better with the order you provide, but i have the feeling that is better to switch the two params order to let it be backward compatible and because default values works better as last parameters

Reply:
As far as backwards compatibility goes I think there shouldn't be that much difference...? :thinking: 

As far as having optional params at the tail of the args I think you are right, but I don't like having the `client` after the `audience` :cry: 

What does @MaeIsBad think?

Reply:
Agreed, the args should be swapped, for some reason my brain thought this code would work completely differently than it actually does

Reply:
Yeah Elixir fills the required args first and then starts filling optional arguments in order, this way optional args can be put wherever (if I understood the docs correctly). I'll swap it around, but I'll be really sad

Reply:
Changed the order, but only for the external API. Internally it was already implicitly ordered like that because the call expects a `GenServer.t()`

Reply:
Internally I think it's fine :+1: